app/src/main/java/com/isc/hermes/model/Searcher.java

```diff
@@ -20,30 +20,95 @@
  * @see CarmenFeature a feature reresented with geojson properties
  */
 public class Searcher {
+
+    protected CurrentLocationModel currentLocationModel;
+
+    /**
+     * This method is the constructor of the class
+     */
+    public Searcher() {
+        currentLocationModel  = new CurrentLocationModel();
+    }
+
+
     /**
-     * Method to get the the suggestions features carmen List to have access of all the properties to render the locations
+     * Retrieves a list of suggestion features for a given user location and query.
      *
-     * @param query the consult of the searcher field text
-     * @return the features list with the suggestions features
+     * @param userLocation The current user location as a CurrentLocationModel object.
+     * @param query        The search query for which suggestion features will be obtained.
+     * @return A list of CarmenFeature objects representing the found suggestion features.
+     * @throws IOException If an error occurs during the geocoding call execution.
+     */
+    public List<CarmenFeature> getSuggestionsFeatures(CurrentLocationModel userLocation, String query) {
+        MapboxGeocoding client = buildGeocodingClient(userLocation, query, "BO");
+        Response<GeocodingResponse> geocodingResponse = executeGeocodingCall(client);
+
+        if (!isGeocodingResponseValid(geocodingResponse)) {
+            client = buildGeocodingClient(userLocation, query, null);
+            geocodingResponse = executeGeocodingCall(client);
+        }
+
+        return getFeaturesFromGeocodingResponse(geocodingResponse);
+    }
+
+    /**
+     * Builds a Mapbox Geocoding client based on the provided user location, query, and country.
+     *
+     * @param userLocation The current user location as a CurrentLocationModel object.
+     * @param query        The search query to be used for geocoding.
+     * @param country      The country code to limit geocoding results (optional, can be null).
+     * @return A MapboxGeocoding object configured with the specified parameters.
      */
-    public List<CarmenFeature> getSuggestionsFeatures(String query) {
-        MapboxGeocoding client = MapboxGeocoding.builder()
+    private MapboxGeocoding buildGeocodingClient(CurrentLocationModel userLocation, String query, String country) {
+        MapboxGeocoding.Builder builder = MapboxGeocoding.builder()
                 .accessToken("sk.eyJ1IjoiaGVybWVzLW1hcHMiLCJhIjoiY2xpamxmbnQxMDg2aDNybGc0YmUzcHloaCJ9.__1WydgkE41IAuYtsob0jA")
```

Comment:
Try to use a centralized constant instead hardcode the token because when the token will changes it will be needed to change it in many places

Reply:
In order to implement a functionality it is required to implement ReverseGeocodingCriteria and with this the API 'com.mapbox.mapboxsdk:mapbox-geocoding:5.0.0' must be implemented to the gradle and no matter how much we add it, it does not let us use its API functions.

So it will be implemented in this way until we find another API which is compatible and let us use it in its entirety.

Reply:
- Centralized constant instead hardcode token Fixed in last Commit
![image](https://github.com/CampusDelSaber/hermes/assets/135616980/e0a84edb-3ac5-4e24-8ac3-2fe8b560304d)